benedict/dicts/keylist/keylist_dict.py

```diff
@@ -65,6 +70,8 @@ def get(self, key, default=None):
 
     def _get_by_keys(self, keys, default=None):
         parent, key, _ = keylist_util.get_item(self, keys)
+        if type_util.is_wildcard(key) and type_util.is_list(parent):
```

Comment:
Invert the conditions order (check the item, then the key), just for consistency/hierarchy.